runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -0,0 +1,130 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata.TableName;
+import org.corfudb.runtime.CorfuStoreMetadata.Timestamp;
+
+/**
+ * Created by zlokhandwala on 2019-08-02.
+ */
+public class CorfuStore {
+
+    private final CorfuRuntime runtime;
```

Comment:
can you pass `SequencerView` and `TableRegistry` as parameters instead of runtime?
There are at least 3 disadvantages of using rt:
 - rt is a huge object, it's overweighted GOD object
 - rt brings circular dependencies between every single class using rt
 - harder to write tests if rt is used in code

Reply:
CorfuStore is a single entity which the client instantiates and uses to create tables, perform CRUD, queries, etc. The client should not have visibility into the different parts of the runtime to instantiate the Corfu Store.
The corfuStore also needs the ObjectsView to handle transactions.